test/support/utils.h

```diff
@@ -967,11 +968,19 @@ struct __kernel_name_with_idx
 
 template <int idx, typename KernelParams>
 constexpr auto
-get_new_kernel_params(KernelParams)
+get_new_kernel_params([[maybe_unused]] KernelParams params)
```

Comment:
I am not sure that `[[maybe_unused]]` is the best way here: I think usually this approach applied to return value and local variables.
Instead that we may use some self-made `PARAM_REF` macro to avoid compiler warnings.

Reply:
[maybe_unused](https://en.cppreference.com/w/cpp/language/attributes/maybe_unused) page lists that it can be applied to a function argument.
Anyway, the argument seems unnecessary, because `KernelParams` can be reconstructed. I've removed it.

Reply:
> [maybe_unused](https://en.cppreference.com/w/cpp/language/attributes/maybe_unused) page lists that it can be applied to a function argument.

It looks tricky after a closer look. Function argument use-case is shown only as an example. Thanks for pointing at it.